src/containers/SignIn/SignIn.tsx

```diff
@@ -1,37 +1,84 @@
 import { useState } from "react";
-
-import Input from "../../components/Inputs/Input";
 import CustomizedButton from "../../components/Buttons/CustomizedButton";
 import Logo from "../../components/Logo/Logo";
 import useStyles from "./../../components/Inputs/InputStyles";
+import RandomButton from "../../components/Buttons/RandomButton";
 
 import Card from "@mui/material/Card";
 import ArrowRightAltIcon from "@mui/icons-material/ArrowRightAlt";
-import RandomButton from "../../components/Buttons/RandomButton";
 import { useNavigate } from "react-router-dom";
-import { TextField } from "@material-ui/core";
+import { Snackbar, TextField } from "@material-ui/core";
+import { Alert } from "@mui/material";
+import { addUser } from "../../redux/reducers/userReducer";
+import { addDoc, collection, serverTimestamp } from "firebase/firestore";
+import { db } from "../../firebase";
+import { useDispatch } from "react-redux";
 
 export default function SignIn() {
-  const [randomString, setRandomString] = useState<string>("");
+
+  const [username, setUsername] = useState("");
+  const [open, setOpen] = useState(false);
   const navigate = useNavigate();
   const classes = useStyles();
+  const dispatch = useDispatch();
 
   const generateRandomString = () => {
+
     const characters =
       "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789";
+
     let result = "";
+
     for (let i = 0; i < 8; i++) {
       const randomIndex = Math.floor(Math.random() * characters.length);
       result += characters.charAt(randomIndex);
     }
-    setRandomString(result);
+
```

Comment:
use a random name generator. store some names in an array and pick a random one from there